src/tikzplotlib/_axes.py

```diff
@@ -330,12 +330,27 @@ def _grid(self, obj, data):
         if has_minor_xgrid:
             self.axis_options.append("xminorgrids")
 
+        # Support gridline color and linestyle
+        def gridline_options(line, data):
+            gridline_options = []
+            gridcolor = line.get_color()
+            data, col, _ = _color.mpl_color2xcolor(data, gridcolor)
+            if col != "black":
+                gridline_options.append(col)
+
+            linestyle = mpl_linestyle2pgfplots_linestyle(
+                data, line.get_linestyle(), line=line
+            )
+            if linestyle is not None and linestyle != "solid":
+                gridline_options.append(linestyle)
+
+            return ", ".join(gridline_options)
+
         xlines = obj.get_xgridlines()
         if xlines:
-            xgridcolor = xlines[0].get_color()
-            data, col, _ = _color.mpl_color2xcolor(data, xgridcolor)
-            if col != "black":
-                self.axis_options.append(f"x grid style={{{col}}}")
+            opts = gridline_options(xlines[0], data)
```

Comment:
`xlines[0]` is a Line2D object. I considered defering the drawing to `_line2d.py` but it would have required more structural changes.